File: APIdoc/php/APIDoc.php
<?php

$BASE = "http://api.point.io/api/v2/";

function auth($email, $password, $apiKey){
	global $BASE;
	$bodyMap= array("email"=>$email, "password"=> $password, "apiKey"=> $apiKey);
	$postString = http_build_query($bodyMap, '', '&');

	$ch = curl_init($BASE . "auth.json");
	curl_setopt($ch, CURLOPT_POST, true);
	curl_setopt($ch, CURLOPT_POSTFIELDS, $postString);
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);
}

function list_access_rules($sessionKey){
	global $BASE;
	$ch = curl_init($BASE . "accessrules/list.json");
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);	
}

function list_folders($sessionKey, $folderId){
	global $BASE;
	$paramMap= array("folderId"=>$folderId);
	$paramStr = http_build_query($paramMap, '', '&');

	$ch = curl_init($BASE . "folders/list.json?" . $paramStr);
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);	
}

function file_preview($sessionKey, $folderid, $fileid, $filename){
	global $BASE;
	$paramMap= array("folderid"=>$folderid, "fileid"=>$fileid, "filename"=>$filename);
	$paramStr = http_build_query($paramMap, '', '&');

	$ch = curl_init($BASE . "folders/files/preview.json?" . $paramStr);
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);		
}

function file_download($sessionKey, $folderid, $fileid, $filename){
	global $BASE;
	$paramMap= array("folderid"=>$folderid, "fileid"=>$fileid, "filename"=>$filename);
	$paramStr = http_build_query($paramMap, '', '&');

	$ch = curl_init($BASE . "folders/files/download.json?" . $paramStr);
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);		
}

function file_create_link($sessionKey, $shareid, $fileid, $filename, $remotepath, $containerid){
	global $BASE;
	$paramMap = array("shareid"=>$shareid, "fileid"=>$fileid, "filename"=>$filename, "remotepath"=>$remotepath, "containerid"=>$containerid);
	$paramStr = http_build_query($paramMap, '', '&');

	$ch = curl_init($BASE . "links/create.json");
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_POST, true);
	curl_setopt($ch, CURLOPT_POSTFIELDS, $paramStr);
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);		
}

function file_upload($sessionKey, $folderid, $fileid, $filename, $filecontents){
	global $BASE;
	$params = array("folderid"=>$folderid, "fileid"=>$fileid, "filename"=>$filename, "filecontents"=>"@".$filecontents);

	$ch = curl_init($BASE . "folders/files/upload.json");
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_POST, true);
	curl_setopt($ch, CURLOPT_POSTFIELDS, $params);
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);		
}

function file_checkout($sessionKey, $folderid, $fileid, $filename){
	global $BASE;
	$paramMap= array("folderid"=>$folderid, "fileid"=>$fileid, "filename"=>$filename);
	$paramStr = http_build_query($paramMap, '', '&');

	$ch = curl_init($BASE . "folders/files/checkout.json?" . $paramStr);
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);		
}

function file_checkin($sessionKey, $folderid, $fileid, $filename){
	global $BASE;
	$paramMap= array("folderid"=>$folderid, "fileid"=>$fileid, "filename"=>$filename);
	$paramStr = http_build_query($paramMap, '', '&');

	$ch = curl_init($BASE . "folders/files/checkin.json?" . $paramStr);
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);		
}

function list_storage_types($sessionKey){
	global $BASE;

	$ch = curl_init($BASE . "storagetypes/list.json");
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);	
}

function list_storage_type_params($sessionKey, $siteTypeId){
	global $BASE;

	$ch = curl_init($BASE . "storagetypes/" . $siteTypeId . "/params.json");
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);	
}

function default_flags(){
	$flags = array("enabled" => "1", "loggingEnabled" => "0", "indexingEnabled" => "0", "revisionControl" => "0", "maxRevisions" => "0", "checkinCheckout" => "0");
	return json_encode($flags);
}

function create_storage_site($sessionKey, $siteTypeId, $name, $flags, $siteArgs){
	global $BASE;
	$paramMap = array("siteTypeId"=>$siteTypeId, "name"=>$name, "siteArguments"=>$siteArgs, "flags"=> $flags);
	$paramStr = http_build_query($paramMap, '', '&');

	$ch = curl_init($BASE . "storagesites/create.json");
	curl_setopt($ch, CURLOPT_HTTPHEADER, array("Authorization: " . $sessionKey));
	curl_setopt($ch, CURLOPT_POST, true);
	curl_setopt($ch, CURLOPT_POSTFIELDS, $paramStr);
	curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

	$response = curl_exec($ch);
	curl_close($ch);

	return json_decode($response, true);		
}

function aasort (&$array, $key) {
    $sorter=array();
    $ret=array();
    reset($array);
    foreach ($array as $ii => $va) {
        $sorter[$ii]=$va[$key];
    }
    asort($sorter);
    foreach ($sorter as $ii => $va) {
        $ret[$ii]=$array[$ii];
    }
    $array=$ret;
}

?>

File: APIdoc/ruby/APIDoc.rb
require 'net/http'
require 'rubygems'
require 'json'

class APIDoc

	def initialize(email, password, apiKey)
		@BASE = "http://api.point.io/api/v2/"
		@email = email
		@password = password
		@apiKey = apiKey			
	end		

	def sessionKey(email = @email, password = @password, apiKey = @apiKey)		
		uri = URI(@BASE + "auth.json")
		res = Net::HTTP.post_form(uri, "email" => email, "password" => password, "apiKey" => apiKey)
		JSON.parse(res.body)["RESULT"]["SESSIONKEY"]
	end	

	def listAccessRules(sessionKey)
		uri = URI(@BASE + "accessrules/list.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Get.new(uri.path)
		req["Authorization"] = sessionKey

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end		

	def listFolders(sessionKey, folderId)
		uri = URI(@BASE + "folders/list.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Get.new(uri.path)
		req["Authorization"] = sessionKey
		req.set_form_data({"folderId" => folderId})

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end	

	def filePreview(sessionKey, folderId, fileId, fileName)
		uri = URI(@BASE + "folders/files/preview.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Get.new(uri.path)
		req["Authorization"] = sessionKey
		req.set_form_data({"folderId" => folderId, "fileId" => fileId, "fileName" => fileName})

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end

	def fileDownload(sessionKey, folderId, fileId, fileName)
		uri = URI(@BASE + "folders/files/download.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Get.new(uri.path)
		req["Authorization"] = sessionKey
		req.set_form_data({"folderId" => folderId, "fileId" => fileId, "fileName" => fileName})

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end	

	def fileUpload(sessionKey, folderId, fileId, fileName, filePath)
		uri = URI(@BASE + "folders/files/upload.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Post.new(uri.path)
		req["Authorization"] = sessionKey
		# req.body_stream=File.open(filePath)
		req["Content-Type"] = "multipart/form-data"
		# req.add_field('Content-Length', File.size(filePath))
		req.set_form_data({"folderId" => folderId, "fileId" => fileId, "fileName" => fileName, "filecontents" => "@upload.txt"})

		res = http.request(req)
		JSON.parse(res.body)
	end	

	def fileCreateLink(sessionKey, folderId, fileId, fileName, remotePath, containerId)
		uri = URI(@BASE + "links/create.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Post.new(uri.path)
		req["Authorization"] = sessionKey
		req.set_form_data({"shareid" => folderId, "fileId" => fileId, "fileName" => fileName, "remotePath" => remotePath, "containerId" => containerId})

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end	

	def fileCheckout(sessionKey, folderId, fileId, fileName)
		uri = URI(@BASE + "folders/files/checkout.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Get.new(uri.path)
		req["Authorization"] = sessionKey
		req.set_form_data({"folderId" => folderId, "fileId" => fileId, "fileName" => fileName})

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end	

	def fileCheckin(sessionKey, folderId, fileId, fileName)
		uri = URI(@BASE + "folders/files/checkin.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Get.new(uri.path)
		req["Authorization"] = sessionKey
		req.set_form_data({"folderId" => folderId, "fileId" => fileId, "fileName" => fileName})

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end	

	def listStorageTypes(sessionKey)
		uri = URI(@BASE + "storagetypes/list.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Get.new(uri.path)
		req["Authorization"] = sessionKey

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end	

	def listStorageTypeParams(sessionKey, siteTypeId)
		uri = URI(@BASE + "storagetypes/" + siteTypeId + "/list.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Get.new(uri.path)
		req["Authorization"] = sessionKey

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end	

	def defaultStorageFlags
		{
			"enabled"=> "1",
			"loggingEnabled"=> "0",
			"indexingEnabled"=> "0",
			"revisionControl"=> "0",
			"maxRevisions"=> "0",
			"checkinCheckout"=> "0"
		}
	end

	def createStorageSite(sessionKey, siteTypeId, name, flags, siteArguments = defaultStorageFlags())
		uri = URI(@BASE + "storagesites/create.json")
		http = Net::HTTP.new(uri.host, uri.port)
		req = Net::HTTP::Post.new(uri.path)
		req["Authorization"] = sessionKey
		req.set_form_data({"siteTypeId" => siteTypeId, "name" => name, "flags" => flags, "siteArguments" => siteArguments})

		res = http.request(req)
		JSON.parse(res.body)["RESULT"]["DATA"]
	end
end

email, password, apiKey = "", "", ""

APIDoc = APIDoc.new(email, password, apiKey)
key = APIDoc.sessionKey()
rules = APIDoc.listAccessRules(key)
folders = APIDoc.listFolders(key, rules[2][1])

upload = APIDoc.fileUpload(key, folders[0][16], "test.txt", "test.txt", "upload.txt")
puts upload